presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -260,7 +262,16 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
                         ImmutableSet.of(new RemoveEmptyDelete()) // Run RemoveEmptyDelete after table scan is removed by PickLayout/AddExchanges
                 ));
 
+        builder.add(
+                new IterativeOptimizer(
+                        stats,
+                        ImmutableSet.of(new ApplyDynamicFilters())
+                ));
+
         builder.add(new PredicatePushDown(metadata, sqlParser)); // Run predicate push down one more time in case we can leverage new information from layouts' effective predicate
+
+        builder.add(new RemoveUnsupportedDynamicFilters());
```

Comment:
The algorithm is next : Introduce all possible dynamic filters at any step of the plan and then remove the not yet unsupported ones (the ones which are not right after table scan). For now we are good. But this might be a problem if we moved to iterative optimizer completely before we introduce support for dynamic filters at any step of the plan.